luarules/gadgets/unit_targetting_priorities.lua

```diff
@@ -71,200 +92,357 @@ if gadgetHandler:IsSyncedCode() then
-        [UnitDefNames["corhunt"].id] = "Scouts",
-        [UnitDefNames["tllprob"].id] = "Scouts",
-        [UnitDefNames["tllrsplane"].id] = "Scouts",
-        [UnitDefNames["tllsonpl"].id] = "Scouts",
-        [UnitDefNames["talon_recon"].id] = "Scouts",
-        [UnitDefNames["talon_vigilante"].id] = "Scouts",
+    local setPriorityGeneralAtLeast100k = {
+        id = CMD_SET_PRIORITY,
+        type = CMDTYPE.ICON_MODE,
+        name = 'Set Priority',
+        action = 'setpriority',
+        tooltip = 'Toggle for target priority\nCurrent priority set to only max health greater than 100000',
+        queueing = false,
+        params = {'4', '', '', '', '', 'At Least\n100k'}
     }
 
-    local function shallow_copy(t)
-        local tmp = {}
+    local healthDescending = {
+        key = "health",
+        map = {} -- empty to trigger __index calls in metatable
+    }
+    local healthDescendingMT = {
+        __index = function (t, k)
+            return 1/k
+        end
+    }
+    setmetatable(healthDescending.map, healthDescendingMT)
 
-        for k, v in pairs(t) do
-            tmp[k] = v
+    local healthAtLeast1k = {
+        key = "health",
+        map = {} -- empty to trigger __index calls in metatable
+    }
+    local healthAtLeast1kMT = {
+        __index = function (t, k)
+            return (k < 1000) and PRIORITY_MUST_NOT or PRIORITY_INDIFFERENT
         end
+    }
+    setmetatable(healthAtLeast1k.map, healthAtLeast1kMT)
 
-        return tmp
-    end
+    local healthAtLeast10k = {
+        key = "health",
+        map = {} -- empty to trigger __index calls in metatable
+    }
+    local healthAtLeast10kMT = {
+        __index = function (t, k)
+            return (k < 10000) and PRIORITY_MUST_NOT or PRIORITY_INDIFFERENT
+        end
+    }
+    setmetatable(healthAtLeast10k.map, healthAtLeast10kMT)
 
-    local airCategoriesCached = {}
+    local healthAtLeast100k = {
+        key = "health",
+        map = {} -- empty to trigger __index calls in metatable
+    }
+    local healthAtLeast100kMT = {
+        __index = function (t, k)
+            return (k < 100000) and PRIORITY_MUST_NOT or PRIORITY_INDIFFERENT
+        end
+    }
+    setmetatable(healthAtLeast100k.map, healthAtLeast100kMT)
 
-    function gadget:Initialize()
-        airCategoriesCached = shallow_copy(airCategories)
-    end
+    local generalPriorities = {
+        [NO_PRIORITY_INDEX] = {
+            cmdDesc = setPriorityGeneralNone,
+            map = noPriority
+        },
+        [1] = {
+            cmdDesc = setPriorityGeneralStrongest,
+            map = healthDescending
+        },
+        [2] = {
+            cmdDesc = setPriorityGeneralAtLeast1k,
+            map = healthAtLeast1k
+        },
+        [3] = {
+            cmdDesc = setPriorityGeneralAtLeast10k,
+            map = healthAtLeast10k
+        },
+        [4] = {
+            cmdDesc = setPriorityGeneralAtLeast100k,
+            map = healthAtLeast100k
+        }
+    }

```

Comment:
'general' priorities are not used by any unit definitions atm, should editing the unit definitions be part of this PR as well? In theory it wouldn't hurt to set this priority for all units that do not have 'air' priorities